spec/app/preflight_check_spec.rb

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+# frozen_string_literal: true
+
+require 'app/preflight_check'
+
+describe App::PreflightCheck do
+  describe '.run!' do
+    let(:client) { double }
+
+    before(:each) do
+      allow(described_class).to receive(:client).and_return(client)
+    end
+
+    context 'when Elasticsearch is not running' do
+      let(:cluster) { double }
+
+      it 'should retry multiple times and fail the check' do
+        stub_const('App::PreflightCheck::STARTUP_RETRY_INTERVAL', 1)
+        stub_const('App::PreflightCheck::STARTUP_RETRY_TIMEOUT', 3)
+        allow(client).to receive(:cluster).and_return(cluster)
```

Comment:
I'm not sure if it's a common pattern, but I feel that stubbing/allowing is more readable when it lives in `before(:each)` section, what do you think @wangch079 ?

Reply:
I see `allow` happens in both `before` and `it` and I have no strong preference. I can make it to `before`.

Reply:
It's not a strong preference from my side :)